ksql-cli/src/main/java/io/confluent/ksql/cli/Cli.java

```diff
@@ -309,68 +298,43 @@ private void printKsqlResponse(final RestResponse<KsqlEntityList> response) {
     }
   }
 
-  private void streamResults(final QueryStream queryStream) {
-    final Future<?> queryStreamFuture = queryStreamExecutorService.submit(() -> {
-      for (long rowsRead = 0; limitNotReached(rowsRead) && queryStream.hasNext(); ) {
-        final StreamedRow row = queryStream.next();
-
-        terminal.printStreamedRow(row);
-        if (row.isTerminal()) {
-          break;
-        }
-
-        if (row.getRow().isPresent()) {
-          rowsRead++;
-        }
-      }
-    });
-
-    terminal.handle(Terminal.Signal.INT, signal -> {
-      terminal.handle(Terminal.Signal.INT, Terminal.SignalHandler.SIG_IGN);
-      queryStream.close();
-    });
-
-    try {
-      if (streamedQueryTimeoutMs != null) {
-        try {
-          queryStreamFuture.get(streamedQueryTimeoutMs, TimeUnit.MILLISECONDS);
-        } catch (final TimeoutException exception) {
-          queryStream.close();
-        }
-      }
-
-      queryStreamFuture.get();
-    } catch (final InterruptedException e) {
-      Thread.currentThread().interrupt();
-    } catch (final ExecutionException e) {
-      if (e.getCause() instanceof RuntimeException) {
-        throw (RuntimeException)e.getCause();
-      }
-      throw new RuntimeException(e.getCause());
-    } finally {
-      terminal.writer().println("Query terminated");
-      terminal.flush();
-    }
-  }
-
-  private boolean limitNotReached(final long rowsRead) {
-    return streamedQueryRowLimit == null || rowsRead < streamedQueryRowLimit;
-  }
-
   @SuppressWarnings({"try", "unused"}) // ignored param is required to compile.
   private void handleQuery(
       final String statement,
       final SqlBaseParser.QueryStatementContext query
   ) {
-    final RestResponse<QueryStream> queryResponse =
-        makeKsqlRequest(statement, restClient::makeQueryRequest);
+    final RestResponse<StreamPublisher<StreamedRow>> queryResponse =
+        makeKsqlRequest(statement, restClient::makeQueryRequestStreamed);
 
     if (!queryResponse.isSuccessful()) {
       terminal.printErrorMessage(queryResponse.getErrorMessage());
```

Comment:
Not your code, but I don't see `terminal.flush()` called after this -- is that necessary?

Reply:
I'll add it anyway, no harm.